### ISSUE TITLE
MySQL 8 error MySQL Database Server/Create User

### ISSUE THREAD

**Issue by rmkimathi** [webmin/webmin#1144]
Failed to save user : SQL set password for 'user101'@'localhost' = password('@|cS#lta@g3a') failed : You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'password('@|cS#lta@g3a')' at line 1

webmin 1.930, MySQL version 8.0.18-0ubuntu0, Ubuntu eoan 19.10

MySQL configuration file: /etc/mysql/mysql.conf.d/mysqld.cnf

**Comment by jcameron** [comment]
Does this only happen when you use a password with special characters in it?

**Comment by rmkimathi** [comment]
> Does this only happen when you use a password with special characters in it?

No, this does not only happen with special characters. Is a webmin script syntax error. Looking at mysql 8 documentation:

Apparently the option to create the user when granting privileges has been removed. The correct command for creating a user in MySQL 8 is CREATE USER.

`use mysql;
CREATE USER 'username'@'localhost' IDENTIFIED BY 'password';
GRANT ALL ON *.* TO 'username'@'localhost';
flush privileges;`

**Comment by jcameron** [comment]
Ok, you're right - the correct syntax should be :

`set password for 'user101'@'localhost' = password('@|cS#lta@g3a')`

On which page in Webmin are you setting the password for a MySQL user?

**Comment by rmkimathi** [comment]
Servers/MySQL Database Server/User Permissions/Create new user

> Ok, you're right - the correct syntax should be :
> 
> `set password for 'user101'@'localhost' = '@|cS#lta@g3a'`
> 
> On which page in Webmin are you setting the password for a MySQL user?

Servers/MySQL Database Server/User Permissions/Create new user

**Comment by jcameron** [comment]
If you go to Servers/MySQL Database Server , what is the exact MySQL version shown at the top of the page?

**Comment by rmkimathi** [comment]
> If you go to Servers/MySQL Database Server , what is the exact MySQL version shown at the top of the page?

MySQL version 8.0.18-0ubuntu0

**Comment by thegamerx1** [comment]
also happening debian 10

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
